bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/internal/AudioManagerImpl.java

```diff
@@ -95,32 +95,33 @@ public void play(AudioStream audioStream, String sinkId) {
     }
 
     public void play(AudioStream audioStream, String sinkId, PercentType volume) {
-        if (audioStream != null) {
-            AudioSink sink = getSink(sinkId);
-
-            if (sink != null) {
-                // get current volume
-                PercentType oldVolume = getVolume(sinkId);
-                // set notification sound volume
-                if (volume != null) {
-                    setVolume(volume, sinkId);
-                }
-                if (sink.getSupportedStreams().stream().anyMatch(clazz -> clazz.isInstance(audioStream))) {
-                    try {
-                        sink.process(audioStream);
-                    } catch (UnsupportedAudioFormatException | UnsupportedAudioStreamException e) {
-                        logger.warn("Error playing '{}': {}", audioStream, e.getMessage(), e);
-                    }
-                } else {
-                    logger.warn("Failed playing audio stream '{}' as audio doesn't support it.", audioStream);
-                }
-                // restore volume
-                if (oldVolume != null) {
-                    setVolume(oldVolume, sinkId);
+        if (audioStream == null) {
+            throw new NullPointerException("Audio stream cannot be played as it is null.");
+        }
+
+        AudioSink sink = getSink(sinkId);
+        if (sink != null) {
+            // get current volume
+            PercentType oldVolume = getVolume(sinkId);
```

Comment:
You should handle volume only if you process the stream.
So I will suggest to move your "volume" stuff inside the if just below.